### ISSUE TITLE
Support for @nestjs/typeorm 8.1

### ISSUE THREAD

**Issue by jifeon** [stelescuraul/rls#108]
Hi!

Thank you for the lib!

Looks there was a breaking change in `v8.1.0` of @nestjs/typeorm - they have dropped helpers folder. You can see it [here](https://github.com/nestjs/typeorm/tree/e1b4953a005f05d4197d91daa390d63f5b70a58f/lib).

rls.module [depends on](https://github.com/Avallone-io/rls/blob/master/lib/rls.module.ts#L19) one of helpers. Looks like helpers were not considered as part of public API of the lib, since the change was introduced in minor version. `v8.0.3` still works fine.

So this issue is about:
1. Support new version of @nestjs/typeorm
2. Think if we really need to depend on private API of the module.

**Comment by stelescuraul** [comment]
Hi @jifeon, Thanks for raising this issue. 
@nestjs/typeorm since `v8.1.0` is using typeorm `v0.3+`. This library is not _yet_ ready to work with typeorm `v0.3`. I will soon start investigating the changes required to support typeorm `v0.3` and I will release a new version in the following weeks. 
I will keep this issue open and report back in the next weeks.

**Comment by jifeon** [comment]
Thank you @stelescuraul!

Just in case somebody faces the same issue, my tmp fix was adding resolution for `@nestjs/typeorm` module for yarn in package.json, like this:

```json
  "resolutions": {
    "@avallone-io/rls/@nestjs/typeorm": "8.0.3"
  },
```

**Comment by theres** [comment]
@stelescuraul did you made any progress? Nest is already in v9 with a couple of neat features like durable providers.

**Comment by stelescuraul** [comment]
Hi everyone, sorry for the late reply on this. I finally managed to find the time and motivation to start on the update. It turns out it wasn't as complex as I initially thought.

I have made a PR with the relevant changes (https://github.com/Avallone-io/rls/pull/139) and it seems like everything works as expected. 
It is currently in draft as I would like to go a bit more in depth with the new methods from typeorm to see if they all respect the RLS model but everything should be merged in the next couple of days.

As for the `nestjs/typeorm` version, I have currently marked it to be working with `>v8.1.0` but I believe `>v9.0.0` should also work as there's no major changes within nestjs/typeorm library as far as I can tell. I will test with it as well and if everything works, I'll make it available to nestjs/typeorm `>v9.0.0` as well.

@theres Durable providers are indeed a neat feature from Nest. However, I am a bit unsure on how (and even if) we should support it for this in this update. Durable providers require some more thinking, especially about how the policies would work. My suggestion is, if you need the durable providers, open a new issue and lets discuss it after the release.

**Comment by stelescuraul** [comment]
version 2.0.0 has been released :tada:. Closing this issue now.
If anything comes up, please open a new one.
